src/main/java/com/blackwaterpragmatic/numbertotext/service/NumberToTextService.java

```diff
@@ -0,0 +1,193 @@
+package com.blackwaterpragmatic.numbertotext.service;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class NumberToTextService {
+	private static final Integer ONE = 1;
+	private static final Integer TEN = 10;
+	private static final Integer TWENTY = 20;
+	private static final Integer ONE_HUNDRED = 100;
+	private static final Integer ONE_THOUSAND = 1000;
+	private static final Integer ONE_MILLION = 1000000;
+	private static final Integer ONE_BILLION = 1000000000;
+
+	private static final String[] ONES = new String[]{
+			"zero",
+			"one",
+			"two",
+			"three",
+			"four",
+			"five",
+			"six",
+			"seven",
+			"eight",
+			"nine"
+	};
+
+	private static final String[] TEENS = new String[]{
+			"ten",
+			"eleven",
+			"twelve",
+			"thirteen",
+			"fourteen",
+			"fifteen",
+			"sixteen",
+			"seventeen",
+			"eightteen",
+			"nineteen"
+	};
+
+	private static final String[] TENS = new String[]{
+			null,
+			null,
+			"twenty",
+			"thirty",
+			"fourty",
+			"fifty",
+			"sixty",
+			"seventy",
+			"eighty",
+			"ninety"
+	};
+
+	/**
+	 * Note the use of LinkedHashMap. Insertion order is important here.
+	 */
+	private static final Map<Integer, String> NUMBER_GROUPS = new LinkedHashMap<Integer, String>() {
+		{
+			put(ONE_BILLION, "billion");
+			put(ONE_MILLION, "million");
+			put(ONE_THOUSAND, "thousand");
+			put(ONE, "");
+		}
+	};
+
+	public String convert(final String numberToConvert) {
+		final Integer number;
+		try {
+			number = Integer.valueOf(numberToConvert);
+		} catch (@SuppressWarnings("unused") final NumberFormatException e) {
+			return String.format("ERROR: Number must be a whole number in the range of %d to %d", Integer.MIN_VALUE, Integer.MAX_VALUE);
+		}
+
+		final String result;
+		if (0 == number) {
```

Comment:
In hindsight, I'm wondering if this clause is needed.  I think `buildGroupNumberString` may handle the zero condition if I removed the `if (moreNumberGroupsToProcess(number)) {` clause.  Trying this out might be worthwhile as removing these 2 clauses would simplify the code a bit.